include/kllvm/codegen/Decision.h

```diff
@@ -30,8 +30,13 @@ class DecisionCase {
   DecisionNode *child;
 
 public:
-  DecisionCase(KOREObjectSymbol *constructor, DecisionNode *child) :
-    constructor(constructor), child(child) {}
+  DecisionCase(
+    KOREObjectSymbol *constructor, 
+    std::vector<std::string> bindings,
+    DecisionNode *child) :
+      constructor(constructor),
+      bindings(bindings),
+      child(child) {}
```

Comment:
Think we will not want to use strings for bindings. We should create some sort of interned strings that can be compared by pointer and call the class `Var` or similar. Interning the strings is some small overhead, but we can assume that the matching will be done many times.

Reply:
Or, it's not completely clear to me, these bindings may not be used at runtime? If the bindings are not used after codegen, leaving them as strings is probably fine.

Reply:
This is purely an identifier used in the compiler, it compiles at runtime into a local llvm value.